primitives/session/src/lib.rs

```diff
@@ -130,3 +130,15 @@ where
 
 	Ok(())
 }
+
+/// A trait to be notified when the session changes.
+/// This is in contrast to `SessionHandler` which handles events for specific keys.
```

Comment:
What is `SesssionHandler`? I presume that trait migth also need to mvoe here, and then you can link the two properly.

Reply:
> What is `SesssionHandler`? I presume that trait migth also need to mvoe here, and then you can link the two properly.

Will take this up in a follow-up to avoid a lot of changes in this one.